File: Zero Phonon Line with Random Strain/lineshape.c
//lineshape.c
//Accounts for the energy split by random strain
//The effect of energy shift not included

#include <stdio.h>      
#include <math.h>      
double h(int n, double args[n]){
    double x1 = (1-args[0])/args[0];
    double x2 = (1-args[1])/args[1];
    double x3 = (1-args[2])/args[2];
    double x4 = (1-args[3])/args[3];
    double freq = args[4];
    double delta = args[5];
    double gamma = args[6];
    double v1 = args[7];
    double v2 = args[8];
    double xc = args[9];
    

    double J = 1/(args[0]*args[1]*args[2]*args[3]*args[0]*args[1]*args[2]*args[3]);
    double E = sqrt((x1+x2) * (x1+x2) + (x3+x4) * (x3+x4));
    double L1 = exp(-(freq - xc + E)*(freq - xc + E)/2/(delta*delta));
    double L2 = exp(-(freq - xc - E)*(freq - xc - E)/2/(delta*delta));
    double distr = pow(1/(x1*x1/v1/v1+x2*x2/v2/v2+x3*x3/v1/v1+x4*x4/v2/v2+gamma*gamma),2.5);
    double h = (L1+L2)*distr*J;
    return h;
}

File: Zero Phonon Line with Random Strain/lineshape.m
function h = lineshape(t1,t2,t3,t4, x, x0, v1, v2, delta, gamma)
	%x1 = t1/(1-t1);
	%x2 = t2/(1-t2);
	%x3 = t3/(1-t3);
	%x4 = t4/(1-t4);
	%J = 1 / ((1-t1)*(1-t1)*(1-t2)*(1-t2)*(1-t3)*(1-t3)*(1-t4)*(1-t4));


	e1 = (1-t1)/t1;
	e2 = (1-t2)/t2;
	e3 = (1-t3)/t3;
	e4 = (1-t4)/t4;
	J = 1 / ((t1)*(t1)*(t2)*(t2)*(t3)*(t3)*(t4)*(t4));


	
	delta = sqrt( (e1+e2)*(e1+e2) + (e3+e4)*(e3+e4) );
	L1 = exp(-(x - x0 + delta)^2/2/delta/delta);
	L2 = exp(-(x - x0 - delta)^2/2/delta/delta);
	h = (L1+L2)/(e1*e1/v1/v1 + e2*e2/v2/v2 + e3*e3/v1/v1 + e4*e4/v2/v2 + gamma^2)^(2.5)*J; 

